src/Illuminate/Queue/Console/WorkCommand.php

```diff
@@ -49,7 +49,11 @@ public function __construct(Worker $worker)
 	 */
 	public function fire()
 	{
-		if ($this->downForMaintenance() && ! $this->option('daemon')) return;
+		if ($this->downForMaintenance() && ! $this->option('daemon'))
+		{
+			$this->worker->sleep($this->option('sleep'));
+			return;
```

Comment:
That's true. I recall from the time I tried with $this->worker->stop() and didn't work. I'm fixing now.